casepro/msgs/models.py

```diff
@@ -356,7 +356,7 @@ def search(cls, org, user, search):
         all_label_access = user.can_administer(org)
 
         if all_label_access:
-            if folder == MessageFolder.inbox:
+            if folder == MessageFolder.inbox or (folder == MessageFolder.archived and label_id):
```

Comment:
We're using label_id because it only need to filter with labels when you select a label in the Archived submenu, when you click on Archived it shouldn't use the has_labels filter